echopype/convert/parse_base.py

```diff
@@ -69,54 +76,274 @@ def _print_status(self):
+                chunks=chunks,
+                dtype="f8",
+                fill_value=np.nan,  # same as float64
+                write_empty_chunks=False,
+            )
 
-        Parameters
-        ----------
-        data_type: str
-            The key of ``self.ping_data_dict_tx`` to rectangularize
-        """
+            # Fill zarr array with actual data
+            z_arr.set_basic_selection(region, arr)
+
+        # Set dask array from zarr array
+        d_arr = da.from_zarr(z_arr)
+        return d_arr
+
+    def _parse_and_pad_datagram(
+        self,
+        data_type,
+        data_type_shapes: dict = {},
+        raw_type: Literal["transmit", "receive"] = "receive",
+        use_swap: bool = False,
+        zarr_root: Optional[zarr.Group] = None,
+        max_chunk_size: str = "100MB",
+    ) -> None:
+        ping_data_dict = self.ping_data_dict_tx if raw_type == "transmit" else self.ping_data_dict
+
+        # If there's no data, set and skip
+        if data_type_shapes[data_type] is None:
+            no_data_dict = {ch_id: None for ch_id in ping_data_dict[data_type].keys()}
+            ping_data_dict[data_type] = no_data_dict
+            return
+
+        # Set up zarr when using swap
+        # by determining the chunk sizes
+        if use_swap:
+            if zarr_root is None:
+                raise ValueError("zarr_root cannot be None when use_swap is True")
+
+            # Get the final data shape
+            data_shape = data_type_shapes[data_type]
+
+            # Auto chunk on first dimension
+            # since this is ping time
+            chunks = ("auto",) + (data_shape[1],)
+            chunks = auto_chunks(
+                chunks=chunks, shape=data_shape, limit=max_chunk_size, dtype=np.dtype("float64")
+            )
+            chunks = chunks + data_shape[2:]
+            chunks = tuple([c[0] if isinstance(c, tuple) else c for c in chunks])
+
+        # Go through data for each channel
+        for ch_id, arr_list in ping_data_dict[data_type].items():
+            # NO DATA -----------------------------------------------------------
+            if all(
+                (arr is None) or (arr.size == 0) for arr in arr_list
+            ):  # if no data in a particular channel
+                # Skip all together if there's no data
+                ping_data_dict[data_type][ch_id] = None
+                continue
+            # -------------------------------------------------------------------
+
+            # If there are data do the following
+            # Add channel id to list if there's data
+            # for "receive" raw type only
+            if raw_type == "receive":
+                self.ch_ids[data_type].append(ch_id)
+
+            # Pad shorter ping with NaN
+            # do this for each channel
+            # this is the first small expansion
+            padded_arr = self.pad_shorter_ping(arr_list)
+
+            # Multiply power data by conversion factor
```

Comment:
Adding this for our future selves
```suggestion
            # data_type="angle" does not require extra manipulation, so below
            # only handles power and complex data
            
            # Multiply power data by conversion factor
```